angular/projects/spark-angular/src/lib/components/sprk-tabs/sprk-tabs.component.ts

```diff
@@ -0,0 +1,362 @@
+
+    if ($event.keyCode === keys.left || $event.keyCode === keys.up) {
+      this.incrementTab(tabElements, panelElements, this.activeClass, -1);
+    } else if ($event.keyCode === keys.right || $event.keyCode === keys.down) {
+      this.incrementTab(tabElements, panelElements, this.activeClass, 1);
+    } else if ($event.keyCode === keys.tab) {
+      if ($event.target.getAttribute('role') === 'tab') {
+        $event.preventDefault();
+        panelElements[
+          this.getActiveTabIndex(tabElements, this.activeClass)
+        ].focus();
+      }
+    } else if ($event.keyCode === keys.home) {
+      this.goToEndTab(tabElements, panelElements, this.activeClass, -1);
+    } else if ($event.keyCode === keys.end) {
+      this.goToEndTab(tabElements, panelElements, this.activeClass, 1);
+    }
+  }
+
+  /**
+   * @ignore
+   */
+  getClasses(): string {
+    const classArray: string[] = ['sprk-c-Tabs'];
+
+    if (this.additionalClasses) {
+      this.additionalClasses.split(' ').forEach((className) => {
+        classArray.push(className);
+      });
+    }
+
+    return classArray.join(' ');
+  }
+
+  /**
+   * @ignore
+   */
+  getContentRelationships(): void {
+    let tabIDs = [];
+    let panelIDs = [];
+
+    if (this.tabs && this.panels) {
+      this.tabs.forEach((tab, index) => {
+        const tabID = `tabs-${this.componentID}-tab-${index}`;
+        const panelID = `tabs-${this.componentID}-panel-${index}`;
+
+        this.renderer.setAttribute(tab.ref.nativeElement, 'id', tabID);
+        this.renderer.setAttribute(
+          tab.ref.nativeElement,
+          'aria-controls',
+          panelID,
+        );
+        tabIDs.push(tabID);
+        panelIDs.push(panelID);
+      });
+
+      tabIDs = tabIDs.reverse();
+      panelIDs = panelIDs.reverse();
+
+      this.panels.forEach((panel) => {
+        this.renderer.setAttribute(
+          panel.ref.nativeElement,
+          'id',
+          panelIDs.pop(),
+        );
+        this.renderer.setAttribute(
+          panel.ref.nativeElement,
+          'aria-labelledby',
+          tabIDs.pop(),
+        );
+      });
+    }
+  }
+
+  /**
+   * @ignore
+   */
+  ngAfterContentInit(): void {
+    // Accounting for dynamic changes in the component
+    this.panels.changes.subscribe(() => {
+      this.getContentRelationships();
+    });
+
+    this.getContentRelationships();
+  }
+
+  /**
+   * @ignore
+   */
+  ariaOrientation(width, element) {
+    // switch aria-orientation on mobile (based on _tabs.scss breakpoint)
+    if (width <= 736) {
+      element.setAttribute('aria-orientation', 'vertical');
+    } else {
+      element.setAttribute('aria-orientation', 'horizontal');
+    }
+  }
+
+  /**
+   * @ignore
+   */
+  getActiveTabIndex(tabs, activeClass) {
+    let activeIndex = null;
+    tabs.forEach((tab, index) => {
+      if (
+        tab.classList.contains(activeClass || 'sprk-c-Tabs__button--active')
```

Comment:
Is there a reason we're needing to do this check? Looks like we have `activeClass` set to `sprk-c-Tabs__button--active` internally. Is there a way to override it that I'm missing?

Reply:
It can be overridden by calling the function directly, for example in a unit test or if you want to change the tab without clicking it. (screenshot from a support thread)

![image](https://user-images.githubusercontent.com/1424560/104767122-ac46b700-5739-11eb-9ece-88d99c0676a6.png)

I don't know why the active class would ever need to be custom though. I'm not sure why it was added to this function originally.